src/components/input/input.js

```diff
@@ -626,15 +628,37 @@ function mdSelectOnFocusDirective() {
   function postLink(scope, element, attr) {
     if (element[0].nodeName !== 'INPUT' && element[0].nodeName !== "TEXTAREA") return;
 
-    element.on('focus', onFocus);
+    var preventMouseUp = false;
+
+    element
+      .on('focus', onFocus)
+      .on('mouseup', onMouseUp);
 
     scope.$on('$destroy', function() {
-      element.off('focus', onFocus);
+      element
+        .off('focus', onFocus)
+        .off('mouseup', onMouseUp);
     });
 
     function onFocus() {
-      // Use HTMLInputElement#select to fix firefox select issues
-      element[0].select();
+      // Use HTMLInputElement#select to fix firefox select issues.
+      // The debounce is here for Edge's sake, otherwise the selection doesn't work.
+      preventMouseUp = true;
```

Comment:
The `mouseup` happens after the focus. E.g. you click, `focus` fires, `mouseup` fires. I didn't have issues with this while testing.